coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -254,31 +256,73 @@ def extract_documentation_with_markers(content, docstyle_definition):
-                               found in the content.
+    DocBaseClass holds important function which will extract, parse
+    and generates diffs for documentation.
     """
-    docstyle_definition = DocstyleDefinition.load(language, docstyle)
-    return extract_documentation_with_markers(content, docstyle_definition)
+
+    def __init__(self):
+        """
+        Instantiates DocBaseClass with useful variables which are used in
+        its member functions.
+        """
+        self.warning_desc = None
+        self.affected_code = None
+        self.diffs = None
+        self.file = None
+        self.filename = None
+
+    def extraction(self, language, docstyle):
+        """
+        Extracts all documentation texts inside the given source-code-string
+        using the coala docstyle definition files.
+
+        The documentation texts are sorted by their order appearing in
+        ``content``.
+
+        For more information about how documentation comments are
+        identified and extracted, see DocstyleDefinition.doctypes enumeration.
+
+        :param language:           The programming language used.
+        :param docstyle:           The documentation style/tool used
+                                   (e.g. doxygen).
+        :raises FileNotFoundError: Raised when the docstyle definition file
+                                   was not found.
+        :raises KeyError:          Raised when the given language is not
+                                   defined in given docstyle.
+        :raises ValueError:        Raised when a docstyle definition setting
+                                   has an invalid format.
+        :return:                   An iterator returning each
+                                   DocumentationComment found in the content.
+        """
+        docstyle_definition = DocstyleDefinition.load(language, docstyle)
+        return extract_documentation_with_markers(
+                        self.file, docstyle_definition)
+
+    def diff_generation(self, doc_comment, new_comment):
+        """
+        Generates diff between the original doc_comment and it's fix
+        new_comment which are instances of DocumentationComment.
+
+        :param doc_comment:
+            original instance of DocumentationComment
+        :param new_comment:
+            fix instance of DocumentationComment
+        """
+        diff = Diff(self.file)
+
+        # We need to update old comment positions, as `assemble()`
+        # prepends indentation for first line.
+        old_range = TextRange.from_values(
+            doc_comment.range.start.line,
+            1,
+            doc_comment.range.end.line,
+            doc_comment.range.end.column)
+
+        # Clearing cached assemble() so a fresh one is fetched.
+        new_comment.assemble.cache_clear()
+
+        diff.replace(old_range, new_comment.assemble())
+        self.affected_code = (diff.range(self.filename),)
+        self.diffs = {self.filename: diff}
```

Comment:
Why are diffs not just returned by this method? What's the benefit of storing them in member in this fashion?